services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -68,14 +68,19 @@ def determine_feedback_from_history(feedback_history)
 
     def regex_is_passing?(entry)
       regex_rules.none? do |regex_rule|
-        regex_rule.sequence_type == RegexRule::TYPE_INCORRECT ? Regexp.new(regex_rule.regex_text).match(entry) : !Regexp.new(regex_rule.regex_text).match(entry)
+        is_matching = match_with_case_sensitive(regex_rule.regex_text, regex_rule.case_sensitive, entry)
+        regex_rule.sequence_type == RegexRule::TYPE_INCORRECT ? is_matching : !is_matching
       end
     end
 
     def display_name
       DISPLAY_NAMES[rule_type.to_sym] || rule_type
     end
 
+    private def match_with_case_sensitive(regex, is_case_sensitive, text)
```

Comment:
This is a good use case for default keyword args.

If you change the method signature to 
```ruby
match_with_case_sensitive(regex, text, is_case_sensitive: true)
```

you both communicate the default value and the type of the parameter.